### PR TITLE
Fix small nits

### PR DESCRIPTION
**Pull request by C2Redhat**
Merge three of Mike Snitzer's patches:

dm vdo index: fix various small nits
dm vdo dedupe: fix various small nits
dm vdo slab-depot: fix various small nits

### CODE REVIEW

**Review by raeburn — APPROVED** [review]

File: src/c++/vdo/base/dedupe.c [hunk 1]
```diff
@@ -1471,14 +1471,13 @@ static void finish_writing(struct hash_lock *lock, struct data_vio *agent)
...
-	    lock->registered)
+	if (vdo_is_state_compressed(lock->duplicate.state) && lock->registered) {
 		/*
 		 * Compression means the location we gave in the UDS query is not the location
 		 * we're using to deduplicate.
 		 */
 		lock->update_advice = true;
-
+        }
 	/* If there are any waiters, we need to start deduping them. */
```
Comment (lorelei-sakai) [inline src/c++/vdo/base/dedupe.c:1481]: You removed a blank line here that Mike didn't.

File: src/c++/uds/src/uds/index-session.c [hunk 2]
```diff
@@ -624,9 +624,10 @@ int uds_close_index(struct uds_index_session *index_session)
 	/* Wait for any current index state change to complete. */
```
Comment (lorelei-sakai) [inline src/c++/uds/src/uds/index-session.c:624]: I can't figure out how to make a note there, but dit looks like Mike's change had a change around line 394 of index-session.c that got completely dropped.

**Review by lorelei-sakai — CHANGES_REQUESTED** [review]
These changes are a little different than Mike's changes in a few spots that I've commented. Not sure how that happened.

**Review by lorelei-sakai — APPROVED** [review]
Looks good now.

**Pull request merged**